src/components/styledSelect/styledSelect.tsx

```diff
@@ -86,6 +95,28 @@ const StyledSelect = ({
     }
   }
 
+  const onKeyDown: KeyboardEventHandler = (e) => {
+    if (!inputRef.current || e.key !== 'Enter') return
+
+    const value = inputRef.current.value.trim()
+
+    if (value.toLowerCase() === activeOption?.optionName.toLowerCase()) return
+
+    const existingOption = options.find(
+      ({ optionName }) => optionName.toLowerCase() === value.toLowerCase()
+    )
+
+    if (existingOption) {
+      selectOption(existingOption.optionValue)
+    } else {
+      setHeaderText(inputRef.current.value)
+      onSubmitInput(value)
+      setIsComponentVisible(false)
```

Comment:
This shouldn't be necessary because `inputRef.current.blur()` will have the same effect.